### PR TITLE
Added feature for opening file after exporting to CSV/XLS/JSON & query files

### PR DESCRIPTION
**Pull request by AlexFsmn**
Related to #746.
Added prompt for opening saved file location/file after save.
This fix includes saving of JSON/CSV/Excel & saving of a new file.

The added feature works in two ways.
First:
1. User saves file as JSON/CSV/Excel
2. If the file is of type JSON or CSV, the file will automatically open.
3. Sqlopsstudio will also show notification with two options:
image
Choosing "Open file location" opens the saved file location and selects the saved file.
Choosing "Open file" opens the file in the default associated program.
The option "Open file" will be shown for all types (JSON/CSV/Excel) which brings the user in control of opening any type in a user default program.

I also added parts of this feature when saving files (through file->new query) which will show the following notification:
image
Note that this will only show for new files.

Please share any thoughts as this may require some discussion before going forward with this solution.

### CODE REVIEW

File: src/sql/parts/query/common/resultSerializer.ts [hunk 1]
```diff
@@ -289,6 +291,31 @@ export class ResultSerializer {
...
+	private promptFileSavedNotification(savedFilePath: string) {
+		let label = getBaseLabel(paths.dirname(savedFilePath));
+
+		this._notificationService.prompt(
+			Severity.Info,
+			LocalizedConstants.msgSaveSucceeded + savedFilePath,
+			[{
+				label: nls.localize('openLocation', "Open file location"),
+				run: () => {
+					var action = new ShowFileInFolderAction(savedFilePath, label || paths.sep, this._windowsService);
```
Comment (kburtram) [inline src/sql/parts/query/common/resultSerializer.ts]: nit: please prefer `let` to `var` in typescript.  Thanks!

File: src/vs/workbench/browser/actions/workspaceActions.ts [hunk 2]
```diff
@@ -131,6 +131,29 @@ export class GlobalRemoveRootFolderAction extends Action {
 	}
 }
 
+
+export class ShowFileInFolderAction extends Action {
```
Comment (kburtram) [inline src/vs/workbench/browser/actions/workspaceActions.ts]: in general we try to minimize change to VS Code source code under `vs`  directory to make refreshing their source code easier.  Could you please move these under `sql` somewhere?

File: src/sql/parts/query/services/queryEditorService.ts [hunk 3]
```diff
@@ -161,7 +166,26 @@ export class QueryEditorService implements IQueryEditorService {
 	public onQueryInputClosed(uri: string): void {
 	}
 
+	private promptFileSavedNotification(savedFilePath: string) {
```
Comment (kburtram) [inline src/sql/parts/query/services/queryEditorService.ts]: I like having the open file option for exported files, but I'm not sure about this part of the change.  If you separate into different PRs I think we can take the changes related to opening exported files per easily.  This change would require more design discussion.

**Comment by AlexFsmn** [comment]
Ok. I will try to split these changes as soon as possible.
@kburtram would you give me a pointer to a more suitable location for the changes currently in the vs directory?
